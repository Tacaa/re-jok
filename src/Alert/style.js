import styled from 'styled-components';
import Icon from '../Icon';
import { hexa } from '../globals';

export const StyledIconClose = styled(Icon)`
 text-align: center;
 position: absolute;
 cursor: pointer;
 height: 1.6rem;
 width: 1.6rem;
 line-height: 1.6rem;
 vertical-align: middle;
 top: 1.6rem;
 right: 1.6rem;



 transition: all .2s;

`

export const StyledAlertIcon = styled(Icon)`
  margin-right: 10px;
  height: 1.6rem;
  width: 1.6rem;
  line-height: 1.6rem;
  vertical-align: middle;
`

const returnType = props => {
  if(!props.color){
    switch (props.type) {
      case 'none':
        return `
          color: ${props.theme.text.regular};
          background: ${ ({theme}) => theme.background.default};

        `
      case 'success':
        return`
          color: ${props.theme.success.main};
          background: ${hexa(props.theme.success.main, .28)};

        `
      case 'info':
        return`
          color: ${props.theme.info.main};
          background: ${hexa(props.theme.info.main, .28)};

        `
      case 'warning':
        return`
          color: ${props.theme.warning.main};
          background: ${hexa(props.theme.warning.main, .28)};

        `
      case 'error':
        return`
          color: ${props.theme.error.main};
          background: ${hexa(props.theme.error.main, .28)};

        `
      default:
        return `
          color: ${props.theme.text.regular};
          background: ${ ({theme}) => theme.background.default};

         `

    }
  }
  else {
    return `
    color: ${props.theme.text.default};
    background: ${props.color};
    `
  }

}

export const StyledAlert = styled.div`
  font-size: 1.4rem;
  position: relative;
  display: flex;
  align-items: center;
  padding: 1.5rem 3rem 1.5rem 1.5rem;
  margin-bottom: 20px;
  ${props => returnType(props)};
`

export const StyledAlertTitle = styled.span`
  font-size: 1.6rem;
  color: ${ ({theme}) => theme.text.primary};
`

export const StyledAlertContent =  styled.div`
  font-size: 1.4rem;
`
