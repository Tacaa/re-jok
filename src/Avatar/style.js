import styled, {css} from 'styled-components';

export const baseAvatar = css `
   display: flex;
   flex: none;
   align-self: center;
   align-items: center;
   justify-content: center;
   list-style: none;
   text-transform: uppercase;
   box-sizing: border-box;
   margin-right: 0.8rem;
   background: ${props => props.imgSrc
  ? "transparent"
  : "#bdbdbd"} ;
   color: #fafafa;
   font-size: ${props => props.theme.size[props.size].fontSize};
   line-height: ${props => props.theme.size[props.size].lineHeight};
   width: ${props => props.theme.size[props.size].width};
   height: ${props => props.theme.size[props.size].height};
   padding: 0;

`

export const StyledAvatar = styled.div `
  ${baseAvatar}

  border-radius: ${props => props.shape === "circle" ? '50%' : '.4rem'};

`
export const StyledAvatarImage = styled.img.attrs({
  src: props => props.imgSrc
    ? props.imgSrc
    : "",
  alt: props => props.imgAlt
    ? props.imgAlt
    : ""
})`

  border-radius: inherit;
  width: 100%;
  height: 100%;
  object-fit: cover;
  vertical-align: middle;
  border-style: none;
`

//=============== Avatar Group ==================
export const StyledAvatarGroup = styled.ul `
  list-style: none;
  display: inline-flex;
  margin-right: .8rem;
  margin-left: .8rem;
  pointer-events: auto;
  order: 1;
`

export const StyledAvatarItem = styled.li `
  margin-left: -.8rem;
  width: ${props => props.theme.size[props.size].width};
  height: ${props => props.theme.size[props.size].height};
  pointer-events: auto;
  box-shadow: rgb(255, 255, 255) 0px 0px 0px 2px;
  position: relative;
  border-radius: 50%;
  transition: transform 0.2s ease-in-out;
`
