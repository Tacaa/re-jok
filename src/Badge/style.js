import styled from 'styled-components'


export const StyledBadge = styled.span`
    font-size: 14px;
    line-height: 1.5;
    color: rgba(0,0,0,.65);
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    list-style: none;
    position: relative;
    display: inline-flex;
    flex-direction: column;
    line-height: 1;
    vertical-align: middle;


`

export const StyledSup = styled.sup`
  background: ${props=> props.backgroundColor ? props.backgroundColor : "#f5222d"};
  position: absolute;
  transform: translateX(-50%);
  top: -1rem;
  left: ${props => props.widthChildren ? `${props.widthChildren/10}rem`: 0}
  height: 20px;
  border-radius: 10px;
  min-width: 20px;
  color: #fff;
  line-height: 20px;
  text-align: center;
  padding: ${props => props.count < 10 ? '0 .6rem' : '0 .8rem'};
  font-size: 12px;
  font-weight: 400;
  z-index: 1;
  white-space: nowrap;

  transform-origin: -10% center;


`
