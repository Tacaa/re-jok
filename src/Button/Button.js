// @flow
import * as React from 'react'

import Icon from '../Icon';

import {
  Btn,
  OutlineBtn,
  Label,
  SpinnerButton,
  IconBtn
} from './style';

import { isChild } from '../helpers/typeUtils';

type Props = {
  /** Set button unable to click or not **/
  disabled?: boolean,
  /** Event when user clicked to button **/
  onClick?: Function,
  /** Set button to a link**/
  hrefString?: string,
  /** The size of button . The default size of button is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of button **/
  shape?: 'circle' | 'rounded',
  /** Button can show a loading**/
  loading?: boolean,
  /** Set icon in button**/
  icon?: string,
  /** Add more style to button **/
  style?: Object,
  /** Add more class to button **/
  className?: string,
  /** Primary color of button **/
  primary?: boolean,
  /** Secondary color of button **/
  secondary?: boolean,
  /** Set width of button to 100%**/
  fullWidth?: boolean,
  /** move icon to left or right label. The default is left **/
  iconPosition: 'left' | 'right',
  /** if you don't want use primary or secondary , you can set your color to button **/
  backgroundColor?: string,
  /** Make your button animated**/
  animated?: string,
  /** Change tag name of button ... span, div, a v..v **/
  tag?: string,
  /** Change type of button**/
  variant: 'outline' | 'inverted' | 'raised' | 'icon',
  /** create button with box shadow or not**/
  hasBoxShadow: boolean,
  /** Set type of button **/
  rule: 'button' | 'reset' | 'submit',
  /**Children element can be string, number, HTML node anything you want**/
  children?: any,
}

const defaultProps = {
  disabled: false,
  size: 'default',
  rule: 'button',
  loading: false,
  primary: false,
  secondary: false,
  fullWidth: false,
  iconPosition: 'left',
  backgroundColor: "",
  tag: "button",
  variant: 'raised',
  hrefString: "",
  hasBoxShadow: false
};

class Button extends React.Component<Props> {
  static defaultProps = defaultProps;

  render() {
    const {
      tag,
      hrefString,
      children,
      loading,
      shape,
      size,
      primary,
      secondary,
      icon,
      iconPosition,
      variant,

    } = this.props;

   const ListButton = {
     raised: Btn,
     outline: OutlineBtn,
     inverted: OutlineBtn,
     icon: IconBtn
   }
   const hasChild = !isChild(children);

    const ElementType = !hrefString ? ListButton[variant].withComponent(tag) : ListButton[variant].withComponent('a')

    return (
      <ElementType {...this.props}>
        {
          icon
            ? (
              loading
              ? (<SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>)
              : (<Icon iconPosition={iconPosition} name={icon}/>))
            : ('')
        }
        {loading && !icon && <SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>}
        {
          shape === "circle" || variant === "icon"
            ? (null)
            : (hasChild && <Label>
              {children}
            </Label>)
        }

      </ElementType>
    )
  }
}

export default Button
