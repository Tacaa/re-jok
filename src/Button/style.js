import styled, {css, keyframes} from 'styled-components';

import {Transition, hexa, shadeColor} from '../globals'

export const baseButton = css `
    display: flex;
    flex: none;
    align-self: center;
    align-items: center;
    justify-content: center;
    word-break: keep-all;
    cursor: pointer;
    position: relative;
    outline: none;
    border: none;
    text-decoration: none;
    text-align: center;

    font-size: ${props => props.theme.size[props.size].fontSize};

      div + span,
      span + span {
        margin: 0 .8rem;
      }




// =======================Button=====================


`
const classicAnimated = css `
&:hover {
    transform: translateY(-.3rem);
    box-shadow: 0 1rem 2rem rgba(0,0,0,.2);

    &::after {
        transform: scaleX(1.4) scaleY(1.6);
        opacity: 0;
    }
}


&:active {
    outline: none;
    transform: translateY(-.1rem);
    box-shadow: 0 .5rem 1rem rgba(0,0,0,.2);
}
`

const basicAnimated = css `
&:hover{
  border-color:${props => props.backgroundColor
        ? hexa(props.backgroundColor, 0.9)
        : props.primary
          ? hexa(props.theme.primary.main, 0.9)
          : props.secondary
            ? hexa(props.theme.secondary.main, 0.9)
            : "#fff"};
  box-shadow: ${props => props.backgroundColor
              ? (`0 14px 26px -12px ${hexa(props.backgroundColor, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.backgroundColor, 0.2)}`)

              : props.primary
                ? (`0 14px 26px -12px ${hexa(props.theme.primary.main, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.theme.primary.main, 0.2)}`)

                : props.secondary
                  ? (`0 14px 26px -12px ${hexa(props.theme.secondary.main, 0.42)}, 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px ${hexa(props.theme.secondary.main, 0.2)}`)

                  : (`0 14px 26px -12px rgba(0, 0, 0, 0.42), 0 4px 23px 0 rgba(0,0,0,0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)`)};

}

&:active{
  box-shadow: -.1rem .1rem .4rem rgba(0, 0, 0, 0.4);
}
`

export const Btn = styled.button.attrs({
  href: props => props.hrefString
    ? props.hrefString
    : null,
  type: props => props.tag === "button"
    ? props.rule
    : null,
  disabled: props => props.disabled || props.loading
})`
  ${baseButton}


  background-color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "#fff"};

  color: ${props => (props.backgroundColor || props.primary || props.secondary)
        ? "#fff"
        : "#000000a6"};

    width: ${props => props.fullWidth && props.shape !== "circle"
          ? "100%"
          : "unset"};


  border: .1rem solid rgba(0,0,0,.15);
  border-color: ${props => props.backgroundColor
            ? props.backgroundColor
            : props.primary
              ? props.theme.primary.main
              : props.secondary
                ? props.theme.secondary.main
                : "#d9d9d9"};


  ${
props => !props.shape
  ? css `
    padding: ${props => props.theme.size[props.size].padding};
    border-radius: ${props => props.theme.size[props.size].borderRadius};
    line-height: ${props => props.theme.size[props.size].lineHeight};

      `
  : props.shape === "circle" && props.variant !== 'icon'
    ? css `
      width: ${props => props.theme.size[props.size].width};
      height: ${props => props.theme.size[props.size].height};
      border-radius: 50%;
      padding: 0;
    `
    : css `
        border-radius: 10rem;
        padding: ${props => props.theme.size[props.size].padding};
        line-height: ${props => props.theme.size[props.size].lineHeight};
    `}

     transition: ${Transition.base.on};
${
  props => props.variant === 'icon'
  ? null
  : props.variant === "outline"
  ? null
  : (
    css`
    &:hover{
      background-color: ${props => props.backgroundColor
      ? hexa(props.backgroundColor, 0.9)
      : props.primary
        ? hexa(props.theme.primary.main, 0.9)
        : props.secondary
          ? hexa(props.theme.secondary.main, 0.9)
          : "#fff"};

    }
    `
    )
}

&::before {
position: absolute;
top: -1px;
left: -1px;
bottom: -1px;
right: -1px;
background: #fff;
opacity: .35;
content: "";
border-radius: inherit;
z-index: 1;
-webkit-transition: opacity .2s;
transition: opacity .2s;
pointer-events: none;
display: ${props => props.disabled || props.loading
      ? "block"
      : 'none'};
}

${props => props.variant === "raised"
        ? (
          !props.animated && props.hasBoxShadow
          ? basicAnimated
          : props.animated === "classic"
            ? classicAnimated
            : null)
        : null}

`

//==================== Outline Button ====================
export const OutlineBtn = Btn.extend `
  background-color: #fff;
  border-color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "rgba(0,0,0,.16)"};
  color: ${props => props.backgroundColor
        ? props.backgroundColor
        : props.primary
          ? props.theme.primary.main
          : props.secondary
            ? props.theme.secondary.main
            : "#0000008a"};

  ${
props => props.variant === "outline"
  ? css `
       &:hover{
            border-color: ${props => props.backgroundColor
    ? shadeColor(props.backgroundColor, -15)
    : props.primary
      ? shadeColor(props.theme.primary.main, -15)
      : props.secondary
        ? shadeColor(props.theme.secondary.main, -15)
        : "rgba(0,0,0,.54)"};

            color: ${props => props.backgroundColor
          ? shadeColor(props.backgroundColor, -15)
          : props.primary
            ? shadeColor(props.theme.primary.main, -15)
            : props.secondary
              ? shadeColor(props.theme.secondary.main, -15)
              : "rgba(0,0,0,.68)"};
       }
     `
  : props.variant === "inverted"
    ? (
      css `
         &:hover{
           background-color: ${props => props.backgroundColor
      ? props.backgroundColor
      : props.primary
        ? props.theme.primary.main
        : props.secondary
          ? props.theme.secondary.main
          : "rgba(0,0,0,.54)"};
            color: #fff;
         }
         `)
    : null}

    ${ props => props.hasBoxShadow ?
      basicAnimated : null
    }



`

//=================== Icon Button =======================
export const IconBtn = Btn.extend `
   border: none;
   color: ${props => props.backgroundColor
  ? props.backgroundColor
  : props.primary
    ? props.theme.primary.main
    : props.secondary
      ? props.theme.secondary.main
      : "#0000008a"};
   width: ${props => props.theme.size[props.size].width};
   height: ${props => props.theme.size[props.size].height};
   border-radius: 50%;
   padding: 0;
   background-color: transparent;
   &:hover{
        border-color: ${props => props.backgroundColor
        ? shadeColor(props.backgroundColor, -15)
        : props.primary
          ? shadeColor(props.theme.primary.main, -15)
          : props.secondary
            ? shadeColor(props.theme.secondary.main, -15)
            : "rgba(0,0,0,.54)"};

        color: ${props => props.backgroundColor
              ? shadeColor(props.backgroundColor, -15)
              : props.primary
                ? shadeColor(props.theme.primary.main, -15)
                : props.secondary
                  ? shadeColor(props.theme.secondary.main, -15)
                  : "rgba(0,0,0,.68)"};
   }

`

//==================== Text Button =======================

const animationTextButton = css`
  &:before{
    content: "";
position: absolute;
bottom: 0;
left: 0;
right: 100%;
border-bottom: 1px solid #222;
transition: right .3s ease-out;
  }
  &:hover:before{
    right: 0;
  }
`


export const TextBtn = styled.button.attrs({
  href: props => props.hrefString
    ? props.hrefString
    : null,
  type: props => props.tag === "button"
    ? props.rule
    : null,
  disabled: props => props.disabled || props.loading
})`
 ${baseButton}
 letter-spacing: 0.08rem;
 background: transparent;
 background-image: none;

 color: ${props => props.color
    ? props.color
    : props.primary
      ? props.theme.primary.main
      : props.secondary
        ? props.theme.secondary.main
        : "#000000a6"};

 padding: ${props => props.theme.size[props.size].padding};
 &:hover{

   color: ${props => props.colorWhenHover ?
        hexa(props.colorWhenHover)
        :
         props.color
          ? hexa(props.color, 0.8)
          : props.primary
            ? hexa(props.theme.primary.main, 0.8)
            : props.secondary
              ? hexa(props.theme.secondary.main, 0.8)
              : hexa("#000000a6", 0.8)};

    border-color: transparent;

 }
 &:active{
   color: ${props => props.colorWhenHover ?
        hexa(props.colorWhenHover)
        : props.color
                  ? hexa(props.color, 0.8)
                  : props.primary
                    ? hexa(props.theme.primary.main, 0.8)
                    : props.secondary
                      ? hexa(props.theme.secondary.main, 0.8)
                      : hexa("#000000a6", 0.8)};
border-color: transparent;
 }

${
  props =>
   props.hoverable && animationTextButton
}



`


// =================== Sub Component for Button ========================

export const Label = styled.span `
  display: block;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  align-self: center;
  margin: auto;
`;

const spin = keyframes `
  to {transform: rotate(360deg);}
`;

export const SpinnerButton = styled.div `
display: inline-block;
width: ${props => props.theme.size[props.size].fontSize};
height: ${props => props.theme.size[props.size].fontSize};
min-width: ${props => props.theme.size[props.size].fontSize};
min-height: ${props => props.theme.size[props.size].fontSize};
position: ${props => props.shape === "circle"
  ? ""
  : "relative"};
color: inherit;
flex: 0 0 ${props => props.theme.size[props.size].fontSize};
order: ${props => props.shape !== "circle"
    ? props.iconPosition === "left"
      ? 0
      : 1
    : 0};
  &::before{
    content: '';
  box-sizing: border-box;
  position: absolute;
  top: 50%;
  left: 50%;
  width: ${props => props.theme.size[props.size].fontSize};
  height: ${props => props.theme.size[props.size].fontSize};
  margin-top: calc(${props => props.theme.size[props.size].fontSize} / -2);
  margin-left: calc(${props => props.theme.size[props.size].fontSize} / -2);
  border-radius: 50%;
  border-color: ${props => props.primary || props.secondary
      ? "#fff"
      : "#d9d9d9"};
  border-top: 2px solid;
  border-right: 2px solid transparent;
  animation: ${spin} .6s linear infinite;
  }
`
