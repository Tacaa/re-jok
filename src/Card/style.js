import styled, {css, keyframes} from 'styled-components'

// ================ Card =================
const opacityCard = keyframes `
0% {

    opacity: 0;
}
100% {
    opacity: 1;
}
`

export const StyledCard = styled.div `

  display: flex;
  flex-direction: column;
  border: .1rem solid rgba(0,0,0,0.1);
  border-radius: .5rem;
  max-width: 100%;

  position: relative;
  font-size: 1.4rem;
  color: rgba(0,0,0,.65);
  margin: 0;


  transition: all .3s;
  box-sizing: border-box;
  list-style: none;
  animation: ${opacityCard} .5s;

  

  ${
    props =>
      {
        if(typeof props.numberCard === "number"){
          return css`
            width: ${props => `calc(${ 100 / props.numberCard}% - 2.4rem)`};
          `
        } else if(typeof props.numberCard === "object"){

        }
        else return css`
           width: 33rem;
        `
      }
  }
  ${
props => props.loading
  ? css `

    padding: 1.4rem;
    height: 40rem;
    border-radius: .5rem;
    `
  : css `
      background: #fff;
      height: 100%;
      padding: 0;
    `}




  ${
props => props.hoverable
  ? css `
       &:hover{
         box-shadow: 0 .2rem .8rem rgba(0,0,0,.09);
         border-color: rgba(0,0,0,.09);
       }
     `
  : null}


`
// ================ Card Group ==============
export const StyledCardGroup = styled.div `
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  width: 100%;
  margin: -1.4rem -.8rem;
  ${
props => props.center
  ? css `
       justify-content: center;
     `
  : null}
  > ${StyledCard}{
      margin: 1.4rem 1.2rem;
  }
`

// ================= Card Header ==============
export const StyledCardHeader = styled.div `
  display: flex;
  align-items: center;
  flex: 0 0 auto;
  padding: 1.4rem;

  @media (min-width: 70rem){
    padding-left: 1.6rem;
    padding-right: 1.6rem;
  }
`

export const StyledCardHeaderInfo = styled.div `
   display: flex;
   align-items: center;
   flex: 1 1 70%;

`
export const StyledCardHeaderAction = styled.div `
    flex: 0 0 30%;
    align-self: flex-start;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;

   & > * {
     margin-right: .6rem;
   }

`

export const StyledCardHeaderText = styled.div `
  display: flex;
  flex-direction: column;


`



//=============== Card Media ===============
export const StyledCardMedia = styled.div `
  display: block;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  padding-top: 56.25%;
  background-image: ${props => props.src
  ? `url("${props.src}")`
  : ""};
`

//================ Card Content =============
export const StyledCardContent = styled.div `
   clear: both;
   color: rgba(0,0,0,.68);
   padding: 1.4rem;
   font-size: 1.4rem;
   line-height: 2rem;
   font-weight: 400;

   @media (min-width: 70rem){
     padding-left: 1.6rem;
     padding-right: 1.6rem;
   }
`

const baseTitle = css`
text-overflow: ellipsis;
overflow: hidden;
letter-spacing: -.029rem;
margin-bottom: .5rem;
white-space: nowrap;
`

export const StyledCardContentTitle = styled.div `
   font-size: ${({theme}) => theme && theme.typography.title.fontSize};
   color: ${({theme}) => theme && theme.text.primary};
   font-weight: ${({theme}) => theme && theme.typography.title.fontWeight};
   line-height: ${({theme}) => theme && theme.typography.title.lineHeight};
   ${baseTitle}

`

export const StyledCardContentSubTitle = styled.div `
font-size: ${({theme}) => theme && theme.typography.subtitle.fontSize};
color:  ${({theme}) => theme && theme.text.secondary};
font-weight: ${({theme}) => theme && theme.typography.subtitle.fontWeight};
line-height: ${({theme}) => theme && theme.typography.subtitle.lineHeight};
${baseTitle}
`

//================ Card Action =============
export const StyledCardAction = styled.div `
   display: flex;
   box-sizing: border-box;
   align-items: center;
   padding: .8rem 1rem;


`

export const StyledActionRight = styled.div `
  display: flex;
  flex: 0 0 50%;
  justify-content: flex-start;
  align-items: center;
`

export const StyledActionLeft = styled.div `
  display: flex;
  flex: 0 0 50%;
  justify-content: flex-end;
  align-items: center;
`
//============== CardLoading =================
const placeHolderShimmer = keyframes `
  0% {
    transform: translateX(-46.8rem);
  }

  100% {
    transform: translateX(46.8rem);
  }
  `
export const PlaceHolder = styled.div `
  width: 100%;
  overflow: hidden;
  height: 21.9rem;
  background:#fafafa;
  position: relative;
  &:before{
    content: "";
    background-image: linear-gradient(
      to right,
      rgb(246, 247, 249) 0%,
      rgb(233, 235, 238) 20%,
      rgb(246, 247, 249) 40%,
      rgb(246, 247, 249) 100%
    );
    background-repeat: no-repeat;
    position: relative;
    width: 100%;
    height: 100%;
    display: block;
    animation: 1s ${placeHolderShimmer} linear infinite;
  }
  & div {
    background: #fff;
    position: absolute;
    right: 0;
  }
  `
const baseAvatarLoading = css `
background: #fafafa !important;
width: 4rem;
height: 4rem;
border: .8rem solid #fafafa;
top: -.8rem;
left: -.8rem;
`
export const AvatarLoading = styled.div `
${baseAvatarLoading}
`

export const AvatarRoundLoading = styled.div `
    ${baseAvatarLoading}
  border-radius: 50%;

  `

export const DividerLoading = styled.div `
  height: ${props => `${props.height / 10}rem`};
  top: ${props => `${props.top / 10}rem`};
  left: ${props => `${props.left / 10}rem`};
`
export const ContentLoading = styled.div `
  height: ${props => `${props.height / 10}rem`};
  top: ${props => `${props.top / 10}rem`};
  left: ${props => `${props.left / 10}rem`};
`

export const NameCardLoading = styled.div `
  height: .6rem;
  top: .8rem;
  left: 13.6rem;
  `
export const DateCardLoading = styled.div `
  height: .6rem;
  top: 2.6rem;
  left: 10rem;
  `
