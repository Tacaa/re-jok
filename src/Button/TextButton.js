// @flow
import * as React from 'react'
import {Icon} from '../Icon';
import {ThemeProvider} from 'styled-components';
import {theme} from '../theme';

import {TextBtn, Label, SpinnerButton} from './style';

type Props = {
  /** Set button unable to click or not **/
  disabled?: boolean,
  /** Event when user clicked to button **/
  onClick?: Function,
  /** Set button to a link**/
  hrefString?: string,
  /** The size of button . The default size of button is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of button **/
  shape?: 'circle' | 'rounded',
  /** Button can show a loading**/
  loading: boolean,
  /** Set icon in button**/
  icon?: string,
  /** Add more style to button **/
  style?: Object,
  /** Add more class to button **/
  className?: string,
  /** Primary color of button **/
  primary?: boolean,
  /** Secondary color of button **/
  secondary?: boolean,
  /** move icon to left or right label. The default is left **/
  iconPosition: 'left' | 'right',
  /** Make your button animated**/
  animated?: string,
  /** Change tag name of button ... span, div, a v..v **/
  tag?: string,

  /**
   * Defines HTML button type attribute
   * @defaultValue 'button'
   */
  rule: 'button' | 'reset' | 'submit',
  /****/
  children?: any,
}


const defaultProps = {
  disabled: false,
  size: 'default',
  rule: 'button',
  loading: false,
  primary: false,
  secondary: false,
  iconPosition: 'left',
  tag: "button",
  color: "",
};

class TextButton extends React.Component<Props> {
  static defaultProps = defaultProps;
  render() {
    const {
      hrefString,
      tag,
      children,
      loading,
      size,
      primary,
      secondary,
      icon,
      iconPosition
    } = this.props;

    const ElementType = !hrefString ? TextBtn.withComponent(tag) : TextBtn.withComponent('a')

    return (<ThemeProvider theme={theme}>
      <ElementType {...this.props}>
        {
          icon
            ? (
              loading
              ? (<SpinnerButton size={size} iconPosition={iconPosition} primary={primary} secondary={secondary}/>)
              : (<Icon iconPosition={iconPosition} name={icon}/>))
            : ('')
        }
        {loading && !icon && <SpinnerButton size={size} iconPosition={iconPosition} primary={primary} secondary={secondary}/>}
        {
           children && <Label>
              {children}
            </Label>
        }

      </ElementType>
    </ThemeProvider>)
  }
}


export default TextButton
