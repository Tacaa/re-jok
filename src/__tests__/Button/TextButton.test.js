import React from 'react';
import { css } from 'styled-components';
import TextButton from '../../Button/TextButton'
import {SpinnerButton} from '../../Button/style'
import {defaultTheme} from '../../theme'
import {Icon} from '../../Icon';
import renderer from 'react-test-renderer'
import { shallowWithTheme, mountWithTheme, renderWithTheme } from '../../helpers/withTheme';
import 'jest-enzyme'
import 'jest-styled-components'

describe('<TextButton />', () => {
  it('should render TextButton', () => {
    const tree = renderWithTheme(
      <TextButton>
        Text Button
      </TextButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a text button correctly', () => {
    const wrapper = mountWithTheme(<TextButton/>, defaultTheme);
    expect(wrapper.find('button').type()).toEqual('button');
 });

 it('renders a button with only text', () => {
   const wrapper = mountWithTheme(<TextButton>Text Button</TextButton>, defaultTheme);
   expect(wrapper.text()).toEqual("Text Button");

   const tree = renderWithTheme(<TextButton>Text Button</TextButton>).toJSON();
   expect(tree).toHaveStyleRule('background', 'transparent');
   expect(tree).toHaveStyleRule('background-image', 'none');
   expect(tree).toHaveStyleRule('font-weight', '600');
   expect(tree).toHaveStyleRule('letter-spacing', '0.08rem');




 });

});
