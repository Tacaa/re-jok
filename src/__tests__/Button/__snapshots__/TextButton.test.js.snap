// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> should render TextButton 1`] = `
.c1 {
  display: block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  line-height: inherit;
  color: inherit;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  word-break: keep-all;
  cursor: pointer;
  position: relative;
  outline: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  font-size: 1.6rem;
  -webkit-letter-spacing: 0.08rem;
  -moz-letter-spacing: 0.08rem;
  -ms-letter-spacing: 0.08rem;
  letter-spacing: 0.08rem;
  background: transparent;
  background-image: none;
  font-weight: 600;
  color: #000000a6;
  padding: 0.8rem 1.2rem;
}

.c0 div + span,
.c0 span + span {
  margin: 0 .8rem;
}

.c0:hover {
  color: rgba(0,0,0,0.8);
  border-color: transparent;
}

.c0:active {
  color: rgba(0,0,0,0.8);
  border-color: transparent;
}

<button
  className="c0"
  color=""
  disabled={false}
  href={null}
  size="default"
  type="button"
>
  
  <span
    className="c1"
  >
    Text Button
  </span>
</button>
`;
